### PR TITLE
Update Freeciv Server

### LINKED ISSUE #216: Update to Freeciv f9bc276f8a63289e7c9806ddb1300ad36687913f
Freeciv version f9bc276f8a63289e7c9806ddb1300ad36687913f added hrm Feature #769078. I don't know if any Freeciv-web side change is required when updating to it. Zeko?

Freeciv commit message:
Change call signature for luascript funcs to allow any return values

All fcdb functions in database.lua have been changed to raise
exceptions rather than return FCDB_ERROR enum, and user_load now returns
a password hash.

See feature #769078

### PR DESCRIPTION
**Pull request by mchenryc**
Updates through `ORIGCAPSTR="+Freeciv.Devel-3.1-2018.Oct.07"`. See commit messages for links.

Note the last commit obsoletes a field in the `PACKET_WEB_CITY_INFO_ADDITION` network packet (See: http://www.hostedredmine.com/issues/789804) Already removed by freeciv/freeciv@d26e02a7.

@kvilhaugsvik Could you take a look at this, as you have made many of the server updates.

**Edit** Added further updates, passing freeciv/freeciv@f9bc276, thus this Closes #216

### CODE REVIEW

**Comment by mchenryc** [comment]
Closing and re-opening to trigger Travis rebuild. Error appears to be ephemeral during apt-get.

**Comment by mchenryc** [comment]
Error appears to be with Travis build - it may go away (such errors that during Oracle JDK update during apt-get are often due to Oracle moving files or removing old versions, later updated by Travis), but updating to the latest Ubuntu (xenial) on Travis also works. When #227 is merged, the build for this PR should succeed.

**Comment by mchenryc** [comment]
@andreasrosdal, @lonemadmax: Maybe we can go ahead and merge this. The changes within this PR are actually minimal, and I would like to see go in before submitting the next server update PR - I'm working on the next, which has a protocol change that requires a few (few yet significant) changes to the .js. I'm still testing that out to be sure I caught all the cases where the change is needed, but hope to submit a PR soon.

File: freeciv-web/src/main/webapp/javascript/fc_types.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ var TRI_MAYBE = 2;
 var MAX_NUM_ITEMS = 200;
 var MAX_NUM_ADVANCES = 250;
 var MAX_NUM_UNITS = 250;
+var MAX_NUM_BUILDINGS = 250;
```
Comment (lonemadmax) [inline freeciv-web/src/main/webapp/javascript/fc_types.js]: Not really a bug, as it's not used in the client (neither this nor B_LAST), but it's 200 there, like MAX_NUM_ITEMS.

**Comment by lonemadmax** [comment]
No one with better knowledge has said anything about #226 in one direction or the other, and undirected tests have not shown anything.

**Pull request merged**